subsys/fs/littlefs_fs.c

```diff
@@ -647,13 +647,30 @@ static lfs_size_t get_block_size(const struct flash_area *fa)
 #endif
 }
 
+/*
+ * NOTE TO USERS:
+ * .storage_dev must be a casted partition ID like:
+ *   .storage_dev = (void *)FIXED_PARTITION_ID(storage_partition)
+ *
+ * Passing a `device *` pointer here (e.g. from device_get_binding)
+ * will NOT work — it will be interpreted as a numeric flash_area ID.
+ *
+ * Valid flash_area IDs are in the range [0, 255].
```

Comment:
You can not assume that. Aside from range, 0x00000020 may be a valid pointer, and it fits in the range.